terraform/aks/data.tf

```diff
@@ -0,0 +1,9 @@
+data "azurerm_key_vault" "app_key_vault" {
+  name                = var.app_key_vault_name
+  resource_group_name = local.app_resource_group_name
+}
+
+data "azurerm_key_vault" "infra_key_vault" {
+  name                = var.infra_key_vault_name
+  resource_group_name = local.app_resource_group_name
+}
```

Comment:
I think you can remove these (and the associated variables) as they're not being used by anything.